src/backend/c-lisp.py

```diff
@@ -28,13 +28,24 @@ def __init__(self):
         self.scopes = []
         # Function name > (ret-type, (arg-types...))
         self.function_types = {}
+        # String literals
+        self.string_literals = {}
 
     def c_lisp(self, prog):
         """Entry point to C-Lisp compiler"""
         if not prog[0] == "c-lisp":
             raise CodegenError("Input not a C-Lisp program")
 
-        return ["brilisp"] + [self.gen_function(fn) for fn in prog[1:]]
+        brilisp_prog = ["brilisp"]
+        for defn in prog[1:]:
+            if defn[0] == "define":
+                brilisp_prog.append(self.gen_function(defn))
+            else:
+                raise CodegenError(f"Not a function: {defn}")
+        brilisp_prog.extend(
```

Comment:
instead of extending add it to the top for readability.

Reply:
Done